### PR TITLE
expression: set maxScale for type Decimal if the scale is NotFixScale

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tics/issues/3373

Problem Summary:

### What is changed and how it works?
Set the scale to MaxDecimalScale if the 

### Check List

Tests 

- [x] Integration test
test case is added in https://github.com/pingcap/tics/pull/3393

Side effects
N/A

Documentation
N/A

### Release note

```release-note
Fix the problem that tiflash may raise the error `DB::Exception: Decimal(48,31)is out of bound`.
```

### CODE REVIEW

**Review by wshwsh12 — APPROVED** [review]
LGTM

File: expression/expression.go [hunk 1]
```diff
@@ -1388,6 +1388,11 @@ func PropagateType(evalType types.EvalType, args ...Expression) {
 				newCol.(*CorrelatedColumn).RetType = col.RetType.Clone()
 				args[0] = newCol
 			}
+			if args[0].GetType().Tp == mysql.TypeNewDecimal {
+				if newDecimal > mysql.MaxDecimalScale {
+					newDecimal = mysql.MaxDecimalScale
+				}
+			}
```
Comment (guo-shaoge) [inline expression/expression.go:1395]: Do we need some testcase to cover it?

Comment (XuHuaiyu) [reply expression/expression.go:1395]: https://github.com/pingcap/tics/pull/3393

**Review by guo-shaoge — APPROVED** [review]

**Pull request merged**